src/test/java/com/es/phoneshop/dao/CustomProductDaoTest.java

```diff
@@ -137,4 +138,82 @@ public void givenIdOfExistingProduct_whenDelete_thenRemoveProduct() {
     public void givenNullId_whenDelete_thenThrowIllegalArgumentException() {
         productDao.delete(null);
     }
+
+    @Test
+    public void givenNullDescription_whenFindProductsByDescription_thenReturnAllProducts() {
+        List<Product> expectedProducts = new ArrayList<>(productDao.getProductList());
+
+        List<Product> actualProducts = productDao.findProductsByDescription(null);
+
+        assertEquals(expectedProducts, actualProducts);
```

Comment:
order of objects

Reply:
doesn't assertEquals check order of lists?